swerve_drive_robot:
  joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 100
  # 주행 모터 컨트롤러 (1~4번 모터, RPM 제어)
  front_left_wheel_joint_controller:
    joint: front_left_wheel_joint
    type: velocity_controllers/JointVelocityController
    pid: {p: 100.0, i: 0.1, d: 10.0}
  front_right_wheel_joint_controller:
    joint: front_right_wheel_joint
    type: velocity_controllers/JointVelocityController
    pid: {p: 100.0, i: 0.1, d: 10.0}
  rear_left_wheel_joint_controller:
    joint: rear_left_wheel_joint
    type: velocity_controllers/JointVelocityController
    pid: {p: 100.0, i: 0.1, d: 10.0}
  rear_right_wheel_joint_controller:
    joint: rear_right_wheel_joint
    type: velocity_controllers/JointVelocityController
    pid: {p: 100.0, i: 0.1, d: 10.0}

  # 조향 모터 컨트롤러 (5~8번 모터, Angle 제어)
  front_left_steering_joint_controller:
    joint: front_left_steering_joint
    type: position_controllers/JointPositionController
    pid: {p: 100.0, i: 0.1, d: 10.0}
  front_right_steering_joint_controller:
    joint: front_right_steering_joint
    type: position_controllers/JointPositionController
    pid: {p: 100.0, i: 0.1, d: 10.0}
  rear_left_steering_joint_controller:
    joint: rear_left_steering_joint
    type: position_controllers/JointPositionController
    pid: {p: 100.0, i: 0.1, d: 10.0}
  rear_right_steering_joint_controller:
    joint: rear_right_steering_joint
    type: position_controllers/JointPositionController
    pid: {p: 100.0, i: 0.1, d: 10.0}